spec.html

```diff
@@ -36023,8 +36296,24 @@ <h1>
         </emu-alg>
         <emu-grammar>AtomEscape :: CharacterClassEscape</emu-grammar>
         <emu-alg>
-          1. Let _A_ be CompileToCharSet of |CharacterClassEscape| with argument _rer_.
-          1. Return CharacterSetMatcher(_rer_, _A_, *false*, _direction_).
+          1. Let _cs_ be CompileToCharSet of |CharacterClassEscape| with argument _rer_.
+          1. If _rer_.[[UnicodeSets]] is *false*, or if every element of _cs_ consists of a single character, return CharacterSetMatcher(_rer_, _cs_, *false*, _direction_).
```

Comment:
This has the same "every element" bit I commented on above, but I don't _think_ a CharacterClassEscape can compile to an empty CharSet, so it can stay as is. It's possible I'm missing a way to use a `\p` escape to produce an empty set, though, in which case this should indeed be updated as well.

Reply:
I suppose it’s _hypothetically_ possible if Unicode ever were to change the definition of `Foo` in `\p{Foo}` to now no apply to any characters. So I think you’re right, we could express this in the spec just to cover all bases.

Reply:
Added the same “(including if _cs_ is empty)” here.

Reply:
Done in https://github.com/mathiasbynens/ecma262/pull/27